### PR TITLE
[Master] Frontend aggregation layer

### PR DESCRIPTION
**Pull request by pallavigopi**
API spec changes for frontend aggregation layer.

### CODE REVIEW

File: spec/definitions/AppVersionDelete.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+type: object
```
Comment (tarun91vas) [inline spec/definitions/AppVersionDelete.yaml:1]: Why is app version delete is linked to a template ? And if it's not the same model for create/update or delete. It's better to mention whether this is used for request for response.

File: spec/paths/v1@enterprise@{enterprise_id}@application@{application_id}@.yaml [hunk 2]
```diff
@@ -0,0 +1,70 @@
+get:
```
Comment (tarun91vas) [inline spec/paths/v1@enterprise@{enterprise_id}@application@{application_id}@.yaml:1]: Should we be making changes to old API also. Can we set deprecation notice for this. @tr0n-p46

File: spec/definitions/AppVersionDelete.yaml (hunk 1 above)
Comment (pallavigopi) [reply spec/definitions/AppVersionDelete.yaml:1]: This is a specific edge case @tarun91vas 
@abdulMuneerM  will be able to answer why its the case.

File: spec/paths/v1@enterprise@{enterprise_id}@application@{application_id}@.yaml (hunk 2 above)
Comment (pallavigopi) [reply spec/paths/v1@enterprise@{enterprise_id}@application@{application_id}@.yaml:1]: What should we do about this?

File: spec/definitions/ApplicationV1Version.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+title: App Version
```
[the remark this answers is not in the document]
Comment (pallavigopi) [reply spec/definitions/ApplicationV1Version.yaml:1]: Filename or Title? @tarun91vas

[the remark this answers is not in the document]
Comment (tarun91vas) [reply spec/definitions/ApplicationV1Version.yaml:1]: Filename, because that will also be used to reference in the API spec.

**Review by tarun91vas — APPROVED** [review]

**Pull request merged**